Apps/System/music.c

```diff
@@ -0,0 +1,461 @@
+#define RECORD_SWOOSH_DISTANCE 130
+#define RECORD_SWOOSH_SPEED 250
+#define ARM_OFFSET_X 17
+#define ARM_OFFSET_Y 77
+#define ARM_OFFSET GPoint(ARM_OFFSET_X, ARM_OFFSET_Y)
+#define ARM_OFFSET_ANGLE -31
+#define ARM_HOME_ANGLE 0
+#define ARM_START_ANGLE 18
+#define ARM_END_ANGLE 46
+#define ARM_SKIP_SPEED 200
+#define ARM_QUICK_SPEED 100
+#define SKIP_DIRECTION_PREV -1
+#define SKIP_DIRECTION_NEXT 1
+#define PROGRESS_PIXELS_MAX 105
+
+static Window *s_main_window;
+static Layer *s_main_layer;
+static ActionBarLayer *s_action_bar;
+
+static n_GPath *s_arm_base_path_ptr;
+static n_GPath *s_arm_1_path_ptr;
+static n_GPath *s_arm_2_path_ptr;
+static n_GPath *s_arm_head_path_ptr;
+
+static const n_GPathInfo ARM_BASE_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{0, -3}, {0, 3}}
+};
+static const n_GPathInfo ARM_1_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{0, -3}, {0, -3-19}}
+};
+static const n_GPathInfo ARM_2_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{0, -3-19}, {19, -3-19-15}}
+};
+static const n_GPathInfo ARM_HEAD_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{19, -3-19-14}, {19+9, -3-19-15-6}}
+};
+
+static Animation *s_animation_record_ptr;
+static Animation *s_animation_arm_ptr;
+static bool s_animating_disk_change;
+static bool s_animating_arm_change;
+static bool s_is_paused;
+static bool s_was_paused;
+static char *s_artist;
+static char *s_track;
+static GBitmap *s_up_bitmap;
+static GBitmap *s_down_bitmap;
+static GBitmap *s_select_bitmap;
+static GColor s_curr_disk_color;
+static GColor s_next_disk_color;
+static GPoint s_curr_disk_pos;
+static GPoint s_next_disk_pos;
+static int32_t s_progress;
+static int32_t s_length;
+static int32_t s_progress_pixels;
+static int32_t s_skip_value;
+static int32_t s_arm_angle;
+static int32_t s_old_arm_angle;
+static int32_t s_animation_start_arm_angle;
+static int32_t s_animation_end_arm_angle;
+static struct tm s_last_time;
+// TODO remove these variables, they are for demoing only
+static int32_t s_trackcounter;
+static char *artists[] = { "The Beatles", "Deadmau5", "Daniel Ingram" };
+static char *tracks[] = { "Maxwell's Silver Hammer", "Strobe", "Smile" };
+static int32_t length[] = { 207, 637, 203 }; // { 46, 5, 10 };
+
+static void get_new_track(){
+    s_trackcounter++;
+    if(s_trackcounter >= 3) {
+        s_trackcounter = 0;
+    }
+    s_artist = artists[s_trackcounter];
+    s_track = tracks[s_trackcounter];
+    s_length = length[s_trackcounter];
+    s_progress = 0;
+}
+static void get_last_track(){
+    s_trackcounter--;
+    if(s_trackcounter < 0) {
+        s_trackcounter = 2;
+    }
+    s_artist = artists[s_trackcounter];
+    s_track = tracks[s_trackcounter];
+    s_length = length[s_trackcounter];
+    s_progress = 0;
+}
+
+static void music_tick(struct tm *tick_time, TimeUnits tick_units) {
+    if((tick_units & SECOND_UNIT) != 0 && !s_is_paused) {
```

Comment:
space between `if` and `(`

Reply:
also, you can simply write `(tick_units & SECOND_UNIT)` -- no `!= 0` needed there